llvm/lib/Analysis/InlineOrder.cpp

```diff
@@ -216,14 +276,43 @@ class PriorityInlineOrder : public InlineOrder<std::pair<CallBase *, int>> {
     };
   }
 
+  PriorityInlineOrder(FunctionAnalysisManager &FAM, const InlineParams &Params,
+                      const Module &M)
+      : FAM(FAM), Params(Params) {
+    isLess = [&](const CallBase *L, const CallBase *R) {
+      return hasLowerPriority(L, R);
+    };
+
+    for (auto &F : M) {
```

Comment:
Instead of iterating over all of these, can't you directly look at the callgraph? https://llvm.org/doxygen/classllvm_1_1CallGraph.html

Reply:
This might be relevant: https://github.com/IBricchi/llvm-project/blob/main/llvm/lib/Analysis/MLInlineAdvisor.cpp#L104

Reply:
That looks much better, just pushed the change